src/atomate2/qchem/sets/core.py

```diff
@@ -0,0 +1,62 @@
+"""Module defining core QChem input set generators."""
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+
+from atomate2.qchem.sets.base import QCInputGenerator
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class SinglePointSetGenerator(QCInputGenerator):
+    """Generate QChem Single Point input sets."""
+
+    def __init__(self) -> None:
+        super().__init__(job_type="sp", scf_algorithm="diis", basis_set="def2-tzvppd")
```

Comment:
These are currently data classes but don't take advantage of the dataclass paradigm. E.g., they implement their own `__init__` function. Instead, I'd recommend making the `QCInputGenerator` class a dataclass first, and then subclasses can be defined as:

```suggestion
@dataclass
class SinglePointSetGenerator(QCInputGenerator):
    """Generate QChem Single Point input sets."""
    
    job_type: str = "sp"
    algorithm: str = "diis"
    basis_set: str = "def2-tzvppd"
```

Which to me is immediately more obvious to see what has been changed.

Reply:
Thanks for these suggestions @utf ! These were really helpful. I was not aware of the `__post_init__()` functionality  in `dataclasses`. It makes the implementation much more concise.

I refactored the `InputSetGenerator` and they are now consistent with the `InputSetGenerator` in `vasp` or `cpk`.

Adding a few more tests in `drones` and `sets` should push the codecov beyond the target. I will try to get to them soon. Let me know if anything stands out to you in the current implementation for the `InputSetGenerator`